lib/protocol/SqlString.js

```diff
@@ -72,20 +73,41 @@ SqlString.arrayToList = function arrayToList(array, timeZone) {
 };
 
 SqlString.format = function format(sql, values, stringifyObjects, timeZone) {
-  values = values == null ? [] : [].concat(values);
+  if (values == null) {
+    return sql;
+  }
 
-  var index = 0;
-  return sql.replace(/\?\??/g, function(match) {
-    if (index === values.length) {
-      return match;
-    }
+  if (!(values instanceof Array)) {
```

Comment:
Well I'd hope unit tests would enforce that the behavior stays the same, but here's a very informal proof that it will stay the same with these changes:

`values` can be one of:

1) a non-null value
```js
// values = 100
[].concat(values) // -> [100]
[values] // -> [100]
```

2) an array
```js
// values = [1, 2]
[].concat(values) // -> [1, 2]
// In this PR values would be unchanged, so it would remain [1, 2]
```

The only reason `[].concat()` was being used before was because the function previously modified the `values` array, so a copy was needed to not modify the user's array, but that was changed in f39856c8b1edeb275a0daaccafe6b62f35e971a8 so the copy is no longer necessary.